microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -485,16 +552,41 @@ def _run_node_init() -> Tuple[DpdkTestResources, str]:
             Task[Tuple[DpdkTestResources, str]](i, _run_node_init, log)
         )
     if rescind_sriov:
-        # re-enable can take a bit, so we'll divide time by 4
-        # and allow for additional time at the end for re-enable
-        time.sleep(seconds // 4)  # sleep before disabling sriov
-        for node_resources in node_cmd_pairs.keys():
+        time.sleep(10)  # run testpmd for a bit before disabling sriov
+        test_kits = node_cmd_pairs.keys()
+
+        # disable sroiv
+        for node_resources in test_kits:
             node_resources.nic_controller._switch_sriov(enable=False)
 
-        time.sleep(seconds // 4)  # re-enable sriov after another 4th of the runtime
-        for node_resources in node_cmd_pairs.keys():
+        # wait for disable to hit the vm
+        for node_resources in test_kits:
+            if not node_resources.wait_for_dmesg_output("AN_DISABLE", seconds // 3):
+                fail(
+                    "Accelerated Network disable not found in dmesg"
+                    f" before timeout for node {node_resources.node.name}"
+                )
+
+        time.sleep(10)  # let testpmd run with sriov disabled
+
+        # re-enable sriov
+        for node_resources in test_kits:
             node_resources.nic_controller._switch_sriov(enable=True)
 
+        # wait for re-enable to hit vms
+        for node_resources in test_kits:
+            if not node_resources.wait_for_dmesg_output("AN_REENABLE", seconds // 2):
+                fail(
+                    "Accelerated Network re-enable not found "
+                    f" in dmesg before timeout for node  {node_resources.node.name}"
+                )
+
+        time.sleep(15)  # let testpmd run with sriov re-enabled
+
+        # kill the commands to collect the output early and terminate before timeout
+        for node_resources in test_kits:
```

Comment:
Besides this, considering to kill commands in `after_case`. So they will be called even the test case is failed. Refer to Lili's PR, it uses this method to cleanup the system.

Reply:
Oh good thought, I can add that in another PR once this merges